unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,235 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import uuid
+import json
+from unittest.mock import MagicMock, patch
+import pytest
+from redis import StrictRedis
+import d3a_api_client
+from d3a_api_client.redis_client_base import RedisClientBase, RedisAPIException
+
+d3a_api_client.redis_client_base.StrictRedis = MagicMock(spec=StrictRedis)
```

Comment:
I think this was already pointed out in a previous comment. This is not the proper way to mock classes: please use `@patch` within the tests that use `StrictRedis` if you want to create a mock for it.
Mocking it at the module level will affect its functionality for all subsequent tests, and this is not what we want.

Reply:
If you want, you can actually mock it inside the `redis_client_auto_register` fixture, so that it will be applied to all tests.

Reply:
Sure, I will adapt to these changes. I will mock it inside the fixture.